src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IProjectContext.cs

```diff
@@ -7,26 +7,34 @@
 namespace Microsoft.VisualStudio.LanguageServices.ProjectSystem
 {
     /// <summary>
-    /// Project context to initialize properties and items of a Workspace project created with <see cref="IProjectContextFactory.CreateProjectContext(string, string, string, Guid, string, Shell.Interop.IVsHierarchy, CommandLineArguments)"/>. 
+    /// Project context to initialize properties and items of a Workspace project created with <see cref="IProjectContextFactory.CreateProjectContext(string, string, string, Shell.Interop.IVsHierarchy)"/>. 
     /// </summary>
-    internal interface IProjectContext
+    internal interface IProjectContext : IDisposable
     {
         // Project properties.
-        ProjectId Id { get; }
         string DisplayName { get; set; }
         string ProjectFilePath { get; set; }
         Guid Guid { get; set; }
         string ProjectType { get; set; }
+        bool DesignTimeBuildStatus { get; set; }
```

Comment:
> To even instantiate this object, I need to pass you a command-line - I can't even instantiate the language service if the design-time build fails.

I am not sure I understand. Shouldn't we be able to get the correct command line, even if the files/references are broken for the project? I am not sure how legacy project system invokes [SetIntellisenseBuildResult](http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/Interop/IIntellisenseBuildTarget.cs,2217a2b1704724d2,references), but there is possibly some logic we can replicate in CPS to set this flag in future. Note that it is true by default, so we are probably fine never setting it for now.